File: level84/homework/codwars7.py
#https://www.codewars.com/kata/517b2bcf8557c200b8000015/train/python
Position = {'high': 'h', 'low': 'l'}

class Warrior:
    def __init__(self, name):
        self.name = name
        self.health = 100
        self.block = ""
        self.deceased = self.zombie = 0
        
    def attack(self, a, b):
        a.set_health(a.health - (a.block != b and 5 + (b == "h") * 5 + (not a.block) * 5))
    
    def set_health(self, n):
        self.health = max(0, n)
        self.zombie = self.deceased
        self.deceased = not self.health

File: level84/homework/codwars7.js
//https://www.codewars.com/kata/517b2bcf8557c200b8000015/train/javascript
const Position = { high: 'h', low: 'l' };

class Warrior{
  constructor(name){    
    this.name = name;  
    this.health = 100;   
    this.deceased = false;
    this.zombie = false;
  }  
  attack(enemy, position){
    if (enemy.block === position) return;
    const damage = (position === Position.high ? 10 : 5) + (enemy.block ? 0 : 5);
    enemy.setHealth(enemy.health - damage);    
  }  
  setHealth(value){
    if (this.deceased) this.zombie = true;
    this.health = Math.max(0, value);  
    this.deceased = this.health == 0;    
  }
}